### ISSUE TITLE
Run Dockerless?

### ISSUE THREAD

**Issue by cocktailpeanut** [All-Hands-AI/OpenHands#426]
**What problem or use case are you trying to solve?**

Run OpenDevin without Docker

**Describe the UX of the solution you'd like**

I'm curious if the Docker requirement is mandatory, or if it is possible to run it without Docker, manually.

**Additional context**

I want to get it to run locally, but without the docker baggage.

**Comment by jojeyh** [comment]
I advise EXTREME caution if you try this because it might delete files or do unexpected things depending on the task you give.  But if you want to try you should be able to modify the code in (https://github.com/OpenDevin/OpenDevin/blob/main/opendevin/controller/command_manager.py)[Command Manager].  Specifically, you want to change line 9
```
self.shell = DockerInteractive(id="default", workspace_dir=dir, container_image=container_image)
```
so that it uses some other class besides `DockerInteractive`.  A simple example might be the class
```
class BashInteractive:
    def execute(self, cmd: str) -> Tuple[int, str]:
        exit_code = os.system(cmd)
        return exit_code, ""
```
This should run the commands using system `bash`.  The docker will still spin up and record logs when you run the backend and that might throw some errors but should be easy to comment out the code required to get rid of errors.

**Comment by jojeyh** [comment]
As a feature request I don't think that is a priority at the moment.

**Comment by rbren** [comment]
Agree--this is too dangerous. Nothing stopping OpenDevin from responding with `rm -rf /` 😅 Going to close this one

**Comment by neubig** [comment]
It's actually possible to do this now, but it needs to be documented better.

**Comment by enyst** [comment]
This was fixed:
https://docs.all-hands.dev/modules/usage/runtimes#local-runtime
